packages/invoices/src/InboundShipment/DetailView/GeneralTab.tsx

```diff
@@ -35,48 +43,188 @@ const Expand: FC<{ rowData: InboundShipmentItem }> = ({ rowData }) => {
   );
 };
 
-export const GeneralTabComponent: FC<GeneralTabProps<InboundShipmentItem>> = ({
-  onRowClick,
-}) => {
+export const GeneralTabComponent: FC<
+  GeneralTabProps<InboundShipmentItem | InvoiceLine>
+> = ({ onRowClick }) => {
   const { pagination } = usePagination();
   const t = useTranslation('common');
 
-  const { data, sortBy, onSort } = useInboundItems();
-  const activeRows = data?.filter(({ isDeleted }) => !isDeleted) ?? [];
+  const lines = useInboundLines();
+  const { data: items } = useInboundItems();
+  const tableStore = useTableStore();
+  const rows = tableStore.isGrouped ? items : lines;
+
+  const paged = useMemo(
+    () => rows?.slice(pagination.offset, pagination.offset + pagination.first),
+    [rows, pagination.offset, pagination.first]
+  );
 
-  const columns = useColumns(
+  const columns = useColumns<InvoiceLine | InboundShipmentItem>(
```

Comment:
Essentially what I was hoping I could 'abstract' away within the `DataTable` component. The TS just became to hard for me right now. One day.